cmd/embedded-cluster/join.go

```diff
@@ -368,3 +412,69 @@ func runK0sInstallCommand(fullcmd string) error {
 	}
 	return nil
 }
+
+func waitForNode(ctx context.Context, kcli client.Client, hostname string) error {
+	loading := spinner.Start()
+	defer loading.Close()
+	loading.Infof("Waiting for node to join the cluster")
+	if err := kubeutils.WaitForNode(ctx, kcli, hostname); err != nil {
+		return fmt.Errorf("unable to wait for node: %w", err)
+	}
+	loading.Infof("Node has joined the cluster!")
+	return nil
+}
+
+func maybeEnableHA(ctx context.Context, kcli client.Client) error {
+	canEnableHA, err := canEnableHA(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("unable to check if HA can be enabled: %w", err)
+	}
+	if !canEnableHA {
+		return nil
+	}
+	shouldEnableHA := prompts.New().Confirm("Do you want to enable high availability?", false)
+	if !shouldEnableHA {
+		return nil
+	}
+	return enableHA(ctx, kcli)
+}
+
+// canEnableHA checks if high availability can be enabled in the cluster.
+func canEnableHA(ctx context.Context, kcli client.Client) (bool, error) {
+	installation, err := kubeutils.GetLatestInstallation(ctx, kcli)
+	if err != nil {
+		return false, fmt.Errorf("unable to get latest installation: %w", err)
+	}
+	if installation.Spec.HighAvailability {
+		return false, nil
+	}
+	var nodes corev1.NodeList
+	if err := kcli.List(ctx, &nodes); err != nil {
+		return false, fmt.Errorf("unable to list nodes: %w", err)
+	}
+	if len(nodes.Items) < 3 {
+		return false, nil
+	}
```

Comment:
we specifically need three controllers, no?

Reply:
Yes, thanks for that call out!

Reply:
addressed in https://github.com/replicatedhq/embedded-cluster/pull/657/commits/c71e6b0c8059aab4688f85cdf22652f5f09b6794